github/enterprise_actions_permissions.go

```diff
@@ -0,0 +1,152 @@
+// Copyright 2021 The go-github AUTHORS. All rights reserved.
+//
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package github
+
+import (
+	"context"
+	"fmt"
+)
+
+// ActionsPermissionsEnterprise represents a policy for allowed actions in an enterprise.
+//
+// GitHub API docs: https://docs.github.com/en/rest/actions/permissions
+type ActionsPermissionsEnterprise struct {
+	EnabledOrganizations *string `json:"enabled_organizations,omitempty"`
+	AllowedActions       *string `json:"allowed_actions,omitempty"`
+	SelectedActionsURL   *string `json:"selected_actions_url,omitempty"`
+}
+
+func (a ActionsPermissionsEnterprise) String() string {
+	return Stringify(a)
+}
+
+// ActionsEnabledOnEnterpriseOrgs represents all the repositories in an enterprise for which Actions is enabled.
+type ActionsEnabledOnEnterpriseOrgs struct {
+	TotalCount    int             `json:"total_count"`
+	Organizations []*Organization `json:"organizations"`
+}
+
+// GetActionsPermissions gets the GitHub Actions permissions policy for an enterprise.
+//
+// GitHub API docs: https://docs.github.com/en/rest/actions/permissions#get-github-actions-permissions-for-an-enterprise
+func (s *EnterpriseService) GetActionsPermissions(ctx context.Context, enterprise string) (*ActionsPermissionsEnterprise, *Response, error) {
+	u := fmt.Sprintf("enterprises/%v/actions/permissions", enterprise)
+
+	req, err := s.client.NewRequest("GET", u, nil)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	permissions := new(ActionsPermissionsEnterprise)
+	resp, err := s.client.Do(ctx, req, permissions)
+	if err != nil {
+		return nil, resp, err
+	}
+
+	return permissions, resp, nil
+}
+
+// EditActionsPermissions sets the permissions policy in an enterprise.
+//
+// GitHub API docs: https://docs.github.com/en/rest/actions/permissions#set-github-actions-permissions-for-an-enterprise
+func (s *EnterpriseService) EditActionsPermissions(ctx context.Context, enterprise string, actionsPermissionsEnterprise ActionsPermissionsEnterprise) (*ActionsPermissionsEnterprise, *Response, error) {
+	u := fmt.Sprintf("enterprises/%v/actions/permissions", enterprise)
+	req, err := s.client.NewRequest("PUT", u, actionsPermissionsEnterprise)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	p := new(ActionsPermissionsEnterprise)
+	resp, err := s.client.Do(ctx, req, p)
+	if err != nil {
+		return nil, resp, err
+	}
+
+	return p, resp, nil
+}
+
+// ListEnabledOrgsInEnterprise lists the selected organizations that are enabled for GitHub Actions in an enterprise.
+//
+// GitHub API docs: https://docs.github.com/en/rest/actions/permissions#list-selected-organizations-enabled-for-github-actions-in-an-enterprise
+func (s *ActionsService) ListEnabledOrgsInEnterprise(ctx context.Context, owner string, opts *ListOptions) (*ActionsEnabledOnEnterpriseOrgs, *Response, error) {
```

Comment:
Shouldn't this be `EnterpriseService` here instead of `ActionsService`?

Reply:
My intention was for the new `ActionsService.ListEnabledOrgsInEnterprise` to mirror the existing [ActionsService.ListEnabledReposInOrg](https://pkg.go.dev/github.com/google/go-github/v48/github#ActionsService.ListEnabledReposInOrg), etc. since they implement the similar `/enterprises/{enterprise}/actions/permissions/organizations` and `/orgs/{org}/actions/permissions/repositories` endpoints respectively.

It _would_ make more sense to have `ListEnabledOrgsInEnterprise` be an `EnterpriseService` method but should `ListEnabledReposInOrg` and friends be moved to `OrganizationsService` then to match?

Reply:
We used to determine which service an endpoint belonged to using the organization imposed by the official GitHub v3 API documentation.

Now I'm not sure that the decision is as clear-cut with the documentation site re-organization.

I'm wondering if the `/enterprises/...` endpoints should be in the `EnterpriseService`, the `/orgs/...` in the OrganizationsService, etc... but then am hesitant to do a complete shuffle of methods, as that is extremely disruptive to existing users.

Does anyone have thoughts or ideas on how to address this with the least amount of disruption?

Reply:
> I'm wondering if the /enterprises/... endpoints should be in the EnterpriseService, the /orgs/... in the OrganizationsService, etc... but then am hesitant to do a complete shuffle of methods, as that is extremely disruptive to existing users.

In this situation i think its a bit difficult, but if documentation is added, existing users should be able to easily migrate to the new methods. If the code is oriented on the GitHub API itself, it should make it easier for new users to archieve their goal with this library, because they know the structure from playing around with the GitHub API in curl. 
Therefore I think all Services should be named by their API Endpoint whenever possible.

Reply:
OK, I think my prior recommendations were incorrect.  Following the documentation links, the official documentation places these endpoints under the "Actions" services, so let's please keep them there.

It looks like we now have a situation of duplicated code in two different files, so let's please resolve that problem, @algorythmic,  and then we can continue with this PR.